sdk/eventgrid/eventgrid-systemevents/review/eventgrid-systemevents.api.md

```diff
@@ -1743,52 +1843,52 @@ export type KnownSystemEventTypes = keyof SystemEventNameToEventData;
 
 // @public
 export interface MachineLearningServicesDatasetDriftDetectedEventData {
-    baseDatasetId?: string;
-    dataDriftId?: string;
-    dataDriftName?: string;
-    driftCoefficient?: number;
-    endTime: Date;
-    runId?: string;
-    startTime: Date;
-    targetDatasetId?: string;
+    baseDatasetId: string;
+    dataDriftId: string;
+    dataDriftName: string;
+    driftCoefficient: number;
+    endTime: Date | null;
```

Comment:
Why null and not endTime?

Reply:
Could be a codegen change. @MaryGao any insights?  This package is still in preview.

Reply:
This is not introduced by codegen but [spec change](https://github.com/Azure/azure-rest-api-specs/pull/31182/files).

JS treats optionality and null differntly, the former would be asent on the wire and the latter means `null` value on the wire. From the spec's author description I guess here `endTime` could be `null` on the wire and generated code should be correct. @l0lawrence correct me if i am wrong!
![{96A9DE97-32CA-4328-9BB2-56DC06071678}](https://github.com/user-attachments/assets/f1810919-0690-4bac-bc26-57701f920eb6)

Reply:
It's from spec change so expected and fine for this preview package.